src/clr-angular/emphasis/alert/_alert.clarity.scss

```diff
@@ -1,21 +1,58 @@
-// Copyright (c) 2016-2018 VMware, Inc. All Rights Reserved.
+// Copyright (c) 2016-2019 VMware, Inc. All Rights Reserved.
 // This software is released under MIT license.
 // The full license information can be found in LICENSE in the root directory of this project.
 
 @mixin generateAlertType($alertType: 'info') {
-  $propMap: map-get($clr-alert-colors, $alertType);
-  background: map-get($propMap, background);
-  color: map-get($propMap, font);
-  $border-prop: map-get($propMap, border);
+  background: getAlertColor($alertType, bg-color);
+  color: getAlertColor($alertType, color);
+  $border-prop: getAlertColor($alertType, border-color);
 
   @if $border-prop != 'none' {
-    border: $clr-default-borderwidth solid $border-prop;
+    border: $clr-alert-borderwidth solid $border-prop;
   } @else {
     border: none;
   }
 
   .alert-icon {
-    color: map-get($propMap, icon-color);
+    color: getAlertColor($alertType, icon);
+  }
+
+  .alert-actions .dropdown .dropdown-toggle {
+    color: getAlertColor($alertType, action);
+    border-color: getAlertColor($alertType, action);
+  }
+
+  .alert-action,
+  .dropdown-toggle {
+    color: getAlertColor($alertType, action);
+    border-color: getAlertColor($alertType, action);
+
+    &:focus,
+    &:hover,
+    &:active {
+      color: getAlertColor($alertType, action-active);
+      border-color: getAlertColor($alertType, action-active);
+    }
+
+    &:active {
+      box-shadow: 0 1px 0 0 getAlertColor($alertType, action-active) inset;
+    }
+  }
+
+  .close {
+    color: getAlertColor($alertType, close-icon);
+    opacity: getAlertColor($alertType, close-icon-opacity);
+
+    clr-icon {
+      fill: getAlertColor($alertType, close-icon);
+    }
+
+    &:focus,
+    &:hover,
```

Comment:
@hippee-lee 
The alert close icon hover state has been moved to here. Reason: it needs to be in this mixin in order for the close icon (and hover state) to be themeable by type. Previously, close icons were not themeable in this way.